### PR TITLE
Don't keep around additional cpu copy of loaded mesh files

### PR DESCRIPTION
**Pull request by Wumpf**
### What

This is mainly a refactor of `re_renderer`'s model loading pipeline, but as the title (== changelog entry) points out, a nice side-effect that arose culling unnecessary memory usage which may be important for large meshes.

@etaloop's attempt to add a color option to `Asset3D` (see https://github.com/rerun-io/rerun/pull/7458) made it clear that the output of the mesh loaders is really hard to work with:
Prior to this PR, they would eagerly create gpu-sided meshes and then store them alongside an optional (but always filled-out) "cpu mesh" (in essence the unpacked model file).

Now instead all model loading goes to a intermediate `CpuModel` which can be rather easily post processed.
Gpu resources are then created as a separate step, consuming the `CpuModel` (it should be trivial to create a variant that doesn't consume if we need this in the future)

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7824?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7824?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7824)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_renderer/src/importer/cpu_model.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+/// Like [`MeshInstance`], but for CPU sided usage in a [`CpuModel`] only.
+pub struct CpuMeshInstance {
+    pub mesh: CpuModelMeshKey,
+    pub world_from_mesh: glam::Affine3A,
+    // TODO(andreas): Expose other properties we have on [`MeshInstance`].
+}
+
+/// A model as stored on the CPU.
```
Comment (emilk) [inline crates/viewer/re_renderer/src/importer/cpu_model.rs:23]: What is the difference between a model and a mesh?

How does `CpuModel` relate to `CpuMesh`?
Is there a `GpuModel` ?

File: crates/viewer/re_renderer/src/importer/cpu_model.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+slotmap::new_key_type! {
+    /// Key for identifying a cpu mesh in a model.
+    pub struct CpuModelMeshKey;
+}
+
+/// Like [`MeshInstance`], but for CPU sided usage in a [`CpuModel`] only.
+pub struct CpuMeshInstance {
+    pub mesh: CpuModelMeshKey,
+    pub world_from_mesh: glam::Affine3A,
```
Comment (emilk) [inline crates/viewer/re_renderer/src/importer/cpu_model.rs:19]: should these be `pub`?

File: crates/viewer/re_renderer/src/importer/stl.rs [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-
 use itertools::Itertools;
 use smallvec::smallvec;
 use tinystl::StlData;
 
 use crate::{
-    mesh::{self, GpuMesh},
-    renderer::MeshInstance,
-    RenderContext,
+    mesh::{self},
```
Comment (emilk) [inline crates/viewer/re_renderer/src/importer/stl.rs:6]:
```suggestion
    mesh,
```

File: crates/viewer/re_renderer_examples/framework.rs [hunk 4]
```diff
@@ -332,7 +332,10 @@ pub fn load_rerun_mesh(re_ctx: &RenderContext) -> Vec<re_renderer::renderer::Mes
     let mut zipped_obj = zip.by_name("rerun.obj").unwrap();
     let mut obj_data = Vec::new();
     std::io::Read::read_to_end(&mut zipped_obj, &mut obj_data).unwrap();
-    re_renderer::importer::obj::load_obj_from_buffer(&obj_data, re_ctx).unwrap()
+    re_renderer::importer::obj::load_obj_from_buffer(&obj_data, re_ctx)
+        .unwrap()
```
Comment (emilk) [inline crates/viewer/re_renderer_examples/framework.rs:336]: Can we return errors instead please to set a good example? using `anyhow` should be pretty easy, and probaly reduce line count (`?` is shorter than `.unwrap()`)

**Review by emilk — APPROVED** [review]

File: crates/viewer/re_renderer/src/importer/cpu_model.rs (hunk 2 above)
Comment (Wumpf) [reply crates/viewer/re_renderer/src/importer/cpu_model.rs:19]: yes, that makes it easy & convenient to consturct them in mesh loaders. see gltf loader
could make them pub(crate) I guess :shrug:

**Pull request merged**